### PR TITLE
use starlark

### LINKED ISSUE #13: Switch the ETH partitioning test to use the Starlark version of the module

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
Fixes #13

### CODE REVIEW

**Comment by h4ck3rk3y** [comment]
```
GOROOT=/usr/local/opt/go/libexec #gosetup
GOPATH=/Users/gyani/go #gosetup
/usr/local/opt/go/libexec/bin/go test -c -o /private/var/folders/36/z4rh472936l7q80hmv5swll00000gn/T/GoLand/___go_test_github_com_kurtosis_tech_examples_go_network_partitioning_test.test github.com/kurtosis-tech/examples/go-network-partitioning-test #gosetup
/usr/local/opt/go/libexec/bin/go tool test2json -t /private/var/folders/36/z4rh472936l7q80hmv5swll00000gn/T/GoLand/___go_test_github_com_kurtosis_tech_examples_go_network_partitioning_test.test -test.v -test.paniconexit0
=== RUN   TestNetworkPartitioning
time="2022-11-23T17:26:27+05:30" level=info msg="------------ CONNECTING TO KURTOSIS ENGINE ---------------"
time="2022-11-23T17:26:31+05:30" level=info msg="------------ EXECUTING MODULE ---------------"
time="2022-11-23T17:31:14+05:30" level=info msg="------------ STARTING TEST CASE ---------------"
time="2022-11-23T17:31:14+05:30" level=info msg="NODES STATUS -- |  el-client-0         |  el-client-1         |  el-client-2         |  el-client-3         |"
time="2022-11-23T17:31:14+05:30" level=info msg="NODES STATUS -- |  block - hash        |  block - hash        |  block - hash        |  block - hash        |"
time="2022-11-23T17:31:14+05:30" level=info msg="------------ CHECKING ALL NODES ARE IN SYNC AT BLOCK '2' ---------------"
time="2022-11-23T17:31:17+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:20+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:23+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:26+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:29+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:32+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:35+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:38+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:42+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:45+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:48+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:51+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:54+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:31:57+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:00+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:03+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:06+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:09+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:12+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:15+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:18+05:30" level=info msg="NODES STATUS -- |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |  00000 - 0xa27..0c7  |"
time="2022-11-23T17:32:21+05:30" level=info msg="NODES STATUS -- |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |"
time="2022-11-23T17:32:24+05:30" level=info msg="NODES STATUS -- |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |"
time="2022-11-23T17:32:27+05:30" level=info msg="NODES STATUS -- |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |"
time="2022-11-23T17:32:30+05:30" level=info msg="NODES STATUS -- |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |  00001 - 0x3d8..f0a  |"
time="2022-11-23T17:32:31+05:30" level=info msg="------------ ALL NODES SYNCED AT BLOCK NUMBER '2' ------------"
time="2022-11-23T17:32:31+05:30" level=info msg="NODES STATUS -- |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:31+05:30" level=info msg="------------ VERIFIED ALL NODES ARE IN SYNC BEFORE THE PARTITION ------------"
time="2022-11-23T17:32:31+05:30" level=info msg="------------ INDUCING PARTITION ---------------"
time="2022-11-23T17:32:33+05:30" level=info msg="------------ CHECKING BLOCKS DIVERGE AT BLOCK NUMBER '4' ---------------"
time="2022-11-23T17:32:33+05:30" level=info msg="NODES STATUS -- |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:36+05:30" level=info msg="NODES STATUS -- |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:39+05:30" level=info msg="NODES STATUS -- |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:42+05:30" level=info msg="NODES STATUS -- |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:45+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:48+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:51+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:54+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00002 - 0x096..005  |  00002 - 0x096..005  |"
time="2022-11-23T17:32:57+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:00+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:03+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:06+05:30" level=info msg="NODES STATUS -- |  00003 - 0xe57..3cc  |  00003 - 0xe57..3cc  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:07+05:30" level=info msg="Node 'el-client-0' produced one block after partitioning (block number '4', block hash '0x3a612802b5a4a936704f74d1136e1a8ee0e27529b935dfe02feb53dfd15fa2ec')"
time="2022-11-23T17:33:09+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:12+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:15+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:18+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00003 - 0x24b..817  |  00003 - 0x24b..817  |"
time="2022-11-23T17:33:19+05:30" level=info msg="Node 'el-client-3' produced one block after partitioning (block number '4', block hash '0x74bc77552c8045b5de74e360c55b1ba2cded7816429330821f8119b58ee3c7fb')"
time="2022-11-23T17:33:19+05:30" level=info msg="------------ VERIFIED THAT PARTITIONS BLOCKS DIVERGE ---------------"
time="2022-11-23T17:33:19+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00004 - 0x74b..7fb  |  00004 - 0x74b..7fb  |"
time="2022-11-23T17:33:19+05:30" level=info msg="------------ HEALING PARTITION ---------------"
time="2022-11-23T17:33:20+05:30" level=info msg="------------ PARTITION HEALED ---------------"
time="2022-11-23T17:33:20+05:30" level=info msg="------------ CHECKING ALL NODES ARE BACK IN SYNC AT BLOCK '6' ---------------"
time="2022-11-23T17:33:21+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00004 - 0x74b..7fb  |  00004 - 0x74b..7fb  |"
time="2022-11-23T17:33:24+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00004 - 0x74b..7fb  |  00004 - 0x74b..7fb  |"
time="2022-11-23T17:33:27+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00004 - 0x74b..7fb  |  00004 - 0x74b..7fb  |"
time="2022-11-23T17:33:30+05:30" level=info msg="NODES STATUS -- |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |  00004 - 0x3a6..2ec  |"
time="2022-11-23T17:33:33+05:30" level=info msg="NODES STATUS -- |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |"
time="2022-11-23T17:33:36+05:30" level=info msg="NODES STATUS -- |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |"
time="2022-11-23T17:33:39+05:30" level=info msg="NODES STATUS -- |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |"
time="2022-11-23T17:33:42+05:30" level=info msg="NODES STATUS -- |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |  00005 - 0xd96..25a  |"
time="2022-11-23T17:33:43+05:30" level=info msg="----------- ALL NODES SYNCED AT BLOCK NUMBER '6' -----------"
time="2022-11-23T17:33:43+05:30" level=info msg="NODES STATUS -- |  00006 - 0x470..dcf  |  00006 - 0x470..dcf  |  00006 - 0x470..dcf  |  00006 - 0x470..dcf  |"
time="2022-11-23T17:33:43+05:30" level=info msg="----------- VERIFIED THAT ALL NODES ARE IN SYNC AFTER HEALING THE PARTITION --------------"
--- PASS: TestNetworkPartitioning (401.48s)
PASS

Process finished with the exit code 0

```

File: ethereum-network-partition-test/main_test.go [hunk 1]
```diff
@@ -38,17 +38,16 @@ const (
...
-	ethModuleImage = "kurtosistech/eth2-merge-kurtosis-module:0.7.2"
+	eth2StarlarkModule = "github.com/kurtosis-tech/eth2-module"
 
 	// must be something greater than 4 to have at least 2 nodes in each partition
 	numParticipants = 4
 
 	participantsPlaceholder = "{{participants_param}}"
 	//participantParam        = `{"elType":"geth","elImage":"ethereum/client-go:v1.10.25","clType":"lodestar","clImage":"chainsafe/lodestar:v1.1.0"}`
-	participantParam     = `{"elType":"geth","elImage":"ethereum/client-go:v1.10.25","clType":"lighthouse","clImage":"sigp/lighthouse:v3.1.2"}`
+	participantParam     = `{"el_client_type":"geth","el_client_image":"ethereum/client-go:v1.10.25","cl_client_type":"lighthouse","cl_client_image":"sigp/lighthouse:v3.1.2"}`
```
Comment (mieubrisse) [inline ethereum-network-partition-test/main_test.go:48]: Whoaaa hang on, are Pari/the Ethereum team/the client teams aware of this?? This is now a huge break in all their existing scripts, etc. wrapped around the module!!

**Review by mieubrisse — CHANGES_REQUESTED** [review]
The new eth2-merge module has breaks on the argument names!! We can do this, but ONLY if all the consumers are fine with it/ready for it!!

File: ethereum-network-partition-test/main_test.go (hunk 1 above)
Comment (h4ck3rk3y) [reply ethereum-network-partition-test/main_test.go:48]: Haven't made them aware of this yet. I demo on Friday. The types were originally protobuf types and it made sense to have snake_case in there. Are you against the casing or against the change of names (elType -> el_type -> el_client_type)? What is your preference?

I don't have strong opinions here.

Comment (mieubrisse) [reply ethereum-network-partition-test/main_test.go:48]: the snake case is great if we were in a world with no users already; it's the breaking change that would require all users to snake-case all their input params. if the users are okay with it it's fine by me, but we'll want their assnet

Comment (h4ck3rk3y) [reply ethereum-network-partition-test/main_test.go:48]: how do I talk to all the users? I am talking to Pari on Friday.

Comment (h4ck3rk3y) [reply ethereum-network-partition-test/main_test.go:48]: Paritosh and Marius seem to be fine with it and say that switching shouldn't be a problem.

Comment (h4ck3rk3y) [reply ethereum-network-partition-test/main_test.go:48]: @mieubrisse will still need your approval on this!

**Review by mieubrisse — APPROVED** [review]

**Pull request merged**